Editor/AGS.Editor/Panes/ScriptEditorBase.cs

```diff
@@ -541,11 +541,57 @@ private void GoToDefinition(string structName, string memberName)
             {
                 if (foundInScript.FileName == AGSEditor.BUILT_IN_HEADER_FILE_NAME)
                 {
-                    Factory.GUIController.LaunchHelpForKeyword(_goToDefinition);
+                    if (_goToDefinition == "player")
+                    {
+                        CharactersComponent charactersComponent = Factory.ComponentController.FindComponent<CharactersComponent>();
+                        charactersComponent.ShowPlayerCharacter();
+                    }
+                    else
+                    {
+                        Factory.GUIController.LaunchHelpForKeyword(_goToDefinition);
+                    }
                 }
                 else if (foundInScript.FileName == Tasks.AUTO_GENERATED_HEADER_NAME)
                 {
-                    Factory.GUIController.ShowMessage("This variable is internally defined by AGS and probably corresponds to an in-game entity such as a Character or Inventory Item.", MessageBoxIcon.Information);
+                    if (found is ScriptVariable)
+                    {
+                        ScriptVariable sVar = found as ScriptVariable;
+                        string varType = sVar.Type;
+                        string varName = sVar.VariableName;
+                        if (varType == "InventoryItem")
```

Comment:
Hmm, Component has a base class. Would it be possible to:
- make virtual method SelectByName
- make method FindComponentByTypeName or similar?

in which case all this code would be replaced by something like:
```
Component(?) comp = Factory.ComponentController.FindComponentByName(varType);
if (comp)
     comp.SelectByname(varName);
```

which will also support components created by plugins.

Reply:
I don't know how to do that, it would require some sort of type list so a component can say all things it can handle - Audio has the clips but maybe also the types?

Reply:
I also suggest to rename SelectByName to something that makes it clear what is selected and where, like SelectItemInProjectTree.
The reason for such suggestion is that the Component works a wide range of things (project explorer, panes, some with files), so it has to be certain. For example, maybe we will have a method for opening an panel to edit an item, etc.

Reply:
Also, tbh I doubt `SelectItem` is a good term here, as judging by the implementation, this method not just selects it in a project tree, but opens a pane for preview/edit. The method's name must clearly tell what is going to happen.

Reply:
> I don't know how to do that, it would require some sort of type list so a component can say all things it can handle - Audio has the clips but maybe also the types?

Well, if it's difficult now, perhaps at least make a function that does this selection and returns a component, just to have something that may be called elsewhere if a need appears.

Reply:
About `SelectItem`, what about `ShowItemPaneByName` ?

Reply:
There is a big note there to not modify the IEditorComponent or it will break plugins....

Reply:
I still find it strange to have this list inside ScriptEditorBase class, because this makes a script editor know everything about which component does what; this is why I was proposing to have this selection in some global class, like ComponentController, or other, where would make more sense.

But i can look into adjusting this later myself, if you don't want to bother with this.

Reply:
Uhm, maybe if the component controller iterates in the IEcomponent `_components` and then checks if they are BaseController, and then ask for the script type the component manages, maybe it could work.

Reply:
> Uhm, maybe if the component controller iterates in the IEcomponent _components and then checks if they are BaseController, and then ask for the script type the component manages, maybe it could work.

Hmm, that's actually one possible thing to do this,
but really, if that will be done with this big hardcoded switch - it would be okay for now too. My point is simply that in my opinion this search belongs to a controller class, rather than a script editor class.

Reply:
@ivan-mogilko made a tentative commit with a different approach, see what do you think about it.

Reply:
Yes, I think it's fine; and it does not add anything to external interface, meaning this may be reverted if there's a better idea in the future.

Reply:
Cool, I will mark this conversation as resolved then!